app/auth/suppliers.py

```diff
@@ -1,8 +1,63 @@
-from app.models import Supplier
+from app import db
+from app.models import AuditEvent, AuditTypes, Supplier
 
 
 def get_supplier(code):
     return Supplier.query.filter(
         Supplier.code == code,
         Supplier.status != 'deleted'
     ).first()
+
+
+def valid_supplier(user):
+    if not all(hasattr(user, attr) for attr in ["role", "supplier_code"]):
```

Comment:
what is this line testing?

Reply:
It provides error handling if someone chooses to hit this endpoint from outside the orams app, like from postman or terminal ie where user is anonymous

Reply:
if you have @login_required on your routes that will never be the case